src/app/shared/components/thumbnail/thumbnail.component.ts

```diff
@@ -5,124 +5,60 @@ import {
   ElementRef,
   HostListener,
   DoCheck,
-  Renderer2,
-  NgZone,
-  OnDestroy,
-  AfterViewInit,
   Output,
   EventEmitter,
+  Inject,
+  OnInit,
 } from '@angular/core';
 
 import { debounce } from 'lodash';
-import debug from 'debug';
-
-import { RecordVO, ItemVO } from '@root/app/models';
+import { ItemVO } from '@root/app/models';
 import { DataStatus } from '@models/data-status.enum';
-import * as OpenSeaDragon from 'openseadragon';
-import { ZoomEvent, FullScreenEvent } from 'openseadragon';
-
-const THUMB_SIZES = [200, 500, 1000, 2000];
+import { GetThumbnailInfo } from '@models/get-thumbnail';
 
 @Component({
   selector: 'pr-thumbnail',
   templateUrl: './thumbnail.component.html',
   styleUrls: ['./thumbnail.component.scss'],
 })
-export class ThumbnailComponent implements DoCheck, OnDestroy, AfterViewInit {
-  @Input() item: ItemVO;
-  @Input() maxWidth;
+export class ThumbnailComponent implements OnInit, DoCheck {
+  @Input() public item: ItemVO;
+  @Input() public maxWidth: number | undefined;
 
-  thumbLoaded = false;
+  @Output() public disableSwipe = new EventEmitter<boolean>(false);
+  @Output() public isFullScreen = new EventEmitter<boolean>(false);
+
+  public thumbLoaded = false;
+  public isZip = false;
+  public imageUrl: string | undefined;
 
   private lastItemFolderLinkId: number;
   private lastMaxWidth: number;
 
   private element: Element;
-  private imageElement: Element;
-  private resizableImageElement: Element;
-
-  private initialZoom: number;
 
   private targetThumbWidth: number;
-  private currentThumbWidth = 200;
   private currentThumbUrl: string;
   private dpiScale = 1;
 
   private lastItemDataStatus: DataStatus;
 
-  private debouncedResize;
-  private debug = debug('component:thumbnail');
-
-  public isZip: boolean = false;
-
-  @Input() hideResizableImage: boolean = true;
-  @Output() disableSwipe = new EventEmitter<boolean>(false);
-  @Output() isFullScreen = new EventEmitter<boolean>(false);
-
-  viewer: OpenSeaDragon.Viewer;
+  private debouncedResize: () => void | undefined;
 
   constructor(
     elementRef: ElementRef,
-    private renderer: Renderer2,
-    private zone: NgZone,
+    @Inject('Image') private imageClass: typeof Image,
```

Comment:
This allows us to mock image loading in unit tests. Previously we were actually sending out HTTP requests in tests to a defunct placeholder image generation website. This could still be useful for the HTML test runner and Storybook, but we should be able to run all of our unit tests without relying on the reliability of external hosts or even being connected to the internet at all.